xwiki-platform-core/xwiki-platform-sharepage/xwiki-platform-sharepage-ui/src/main/resources/XWiki/SharePage.xml

```diff
@@ -337,7 +337,7 @@ $!{escapetool.xml($message)}
 &lt;/pre&gt;
 #end
 
-&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${doc.getExternalURL()}&lt;/a&gt; in your browser.&lt;/p&gt;
+&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${util.decodeURI("${doc.getExternalURL()}")}&lt;/a&gt; in your browser.&lt;/p&gt;
```

Comment:
It's not the right way of doing this :) Once the URL is encoded you cannot unencode it... So I think we need to offer an API to not encode the URL, when required.